### PR TITLE
@MirkoPralica Reworked the method GetMammalsByLifeSpan

### PR DESCRIPTION
**Pull request by MirkoPralica**

### CODE REVIEW

File: MammalAPI/Controllers/MammalController.cs [hunk 1]
```diff
@@ -92,7 +92,7 @@ public async Task<IActionResult> GetMammalByLifeSpan(int fromYear, int toYear)
             try
             {
                 var result= await _repository.GetMammalsByLifeSpan(fromYear, toYear);
-                var mappedResult = _mapper.Map<MammalLifespanDTO>(result);
+                var mappedResult = _mapper.Map<List<Mammal>>(result);
```
Comment (Gerglamesh) [inline MammalAPI/Controllers/MammalController.cs:95]: Shouldn't it actually be MammalDTO here? Cause we want to map from a list of Mammal (result) to a list of mammalDTO (mappedResult)?

File: MammalAPI/Services/MammalRepository.cs [hunk 2]
```diff
@@ -69,18 +69,13 @@ public async Task<List<FamilyDTO>> GetMammalsByHabitatId(int id)
             return await query.ToListAsync();
         }
 
-        public async Task<List<MammalLifespanDTO>> GetMammalsByLifeSpan(int fromYear, int toYear)
+        public async Task<List<Mammal>> GetMammalsByLifeSpan(int fromYear, int toYear)
```
Comment (Gerglamesh) [inline MammalAPI/Services/MammalRepository.cs:72]: This is correct though as we are working in the repository and hence work with models.

File: MammalAPI/Controllers/MammalController.cs [hunk 3]
```diff
@@ -92,7 +92,7 @@ public async Task<IActionResult> GetMammalByLifeSpan(int fromYear, int toYear)
             try
             {
                 var result= await _repository.GetMammalsByLifeSpan(fromYear, toYear);
-                var mappedResult = _mapper.Map<MammalLifespanDTO>(result);
+                var mappedResult = _mapper.Map<MammalDTO>(result);
```
Comment (MirkoPralica) [inline MammalAPI/Controllers/MammalController.cs:95]: Fixed.

Comment (singelmamma) [reply MammalAPI/Controllers/MammalController.cs:95]: Nej, det ska vara en lista.

**Pull request merged**